### PR TITLE
RHDM-886: set and override maven-compiler-plugin to version 3.8.1

### PR DESCRIPTION
**Pull request by almope**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -483,8 +483,9 @@
     <version.org.codehaus.plexus.plexus-io>3.0.0</version.org.codehaus.plexus.plexus-io>
     <version.org.apache.maven.shared.maven-artifact-transfer>0.9.1
     </version.org.apache.maven.shared.maven-artifact-transfer>
-    <version.org.apache.maven.plugins.maven-compiler-plugin>3.7.0
+    <version.org.apache.maven.plugins.maven-compiler-plugin>3.8.1
     </version.org.apache.maven.plugins.maven-compiler-plugin>
+    <version.compiler.plugin>3.8.1</version.compiler.plugin>
```
Comment (mareknovotny) [inline pom.xml]: duplicate definition?

Comment (almope) [reply pom.xml]: We need this to override the jboss-parent definition:
https://github.com/jboss/jboss-parent-pom/blob/master/pom.xml#L316

Comment (mareknovotny) [reply pom.xml]: yes, but we have now 2 version properties in kie-parent, so I would recommend to remove one - `version.org.apache.maven.plugins.maven-compiler-plugin` as that is obviously not used or wrongly used ;)

Comment (almope) [reply pom.xml]: Updated to remove `version.org.apache.maven.plugins.maven-compiler-plugin`

**Comment by almope** [comment]
Replaced by https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1027

**Pull request closed without merging**